src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+    grant_ids = []
+    for resp in grant_response_g1:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_ids.append(resp['grantId'])
+
+    del grant_request_g1, grant_response_g1
+
+    # Step 5.1: Sending Heartbeat Request
+    # Construct heartbeat message
+    heartbeat_requests = {'heartbeatRequest': []}
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      heartbeat_requests['heartbeatRequest'].append({
+        'cbsdId': cbsd_id,
+        'grantId': grant_id,
+        'operationState': 'GRANTED'
+      })
+    heartbeat_responses = self._sas.Heartbeat(heartbeat_requests)['heartbeatResponse']
+
+    # Check the heartbeat response code is either 0(SUCCESS) or
+    # 501(SUSPENDED_GRANT)
+    for resp in heartbeat_responses:
+      self.assertIn(resp['response']['responseCode'], [0, 501])
+
+    # Sending relinquishment request for the grant 'G1'.
+    relq_requests = {'relinquishmentRequest': []}
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      relq_requests['relinquishmentRequest'].append({
+        'cbsdId': cbsd_id,
+        'grantId': grant_id
+      })
+    relq_responses = self._sas.Relinquishment(relq_requests)['relinquishmentResponse']
+    self.assertEqual(len(relq_responses), len(relq_requests['relinquishmentRequest']))
+
+    # Check relinquishment response
+    for relq_resp in relq_responses:
+      self.assertEqual(relq_resp['response']['responseCode'], 0)
+
+    del relq_requests, relq_responses
+    del heartbeat_requests, heartbeat_responses
+
+    # TODO
+    # Step 6: Modify the FSS database to include modified version of FSS site S
+
+    # Step 7: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 8: Send Grant Request 'G2'.
+    # Forming of 'G2' grants for all the configured grants
+    addCbsdIdsToRequests(cbsd_ids, config['grantRequests'][1])
+    grant_request_g2 = {'grantRequest': config['grantRequests'][1]}
+
+    grant_response_g2 = self._sas.Grant(grant_request_g2)['grantResponse']
+
+    # Check grant 'G2' response (responseCode should be SUCCESS(0))
+    grant_ids = []
+    for resp in grant_response_g2:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_ids.append(resp['grantId'])
+
+    del grant_request_g2, grant_response_g2
+
+    # Step 9: Sending Heartbeat Request
+    # Construct heartbeat message
+    heartbeat_requests = {'heartbeatRequest': []}
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      heartbeat_requests['heartbeatRequest'].append({
+        'cbsdId': cbsd_id,
+        'grantId': grant_id,
+        'operationState': 'GRANTED'
+      })
+    heartbeat_responses = self._sas.Heartbeat(heartbeat_requests)['heartbeatResponse']
+
+    # Check the heartbeat response code is either 0(SUCCESS) or
+    # 501(SUSPENDED_GRANT)
+    for resp in heartbeat_responses:
+      self.assertIn(resp['response']['responseCode'], [0, 501])
```

Comment:
Same here (configurable).

Reply:
FDB4.     Used the expectedResponseCodes from the config file during checking   heartbeat response.